### PR TITLE
Upload torch dynamo perf stats to Rockset

### PR DESCRIPTION
**Pull request by huydhn**
The new workflow is run after `inductor` or `inductor-perf-test-nightly` finish in trunk (not on PR).  All test reports CSV files are ingested into https://console.rockset.com/collections/details/inductor.torch_dynamo_perf_stats

### Testing

Run

* inductor-A100-perf
```
python -m tools.stats.upload_dynamo_perf_stats --workflow-run-id 4272892998 --workflow-run-attempt 1 --repo pytorch/pytorch
```

to ingest some data from https://hud.pytorch.org/pytorch/pytorch/commit/9b7abc4facea1dff69f81e2819570afcd36b35f5

### CODE REVIEW

File: .github/workflows/upload-torch-dynamo-perf-stats.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: Upload torch dynamo performance stats
+
+on:
+  workflow_run:
+    workflows: [inductor, inductor-A100-perf, periodic]
```
Comment (desertfire) [inline .github/workflows/upload-torch-dynamo-perf-stats.yml:5]: We only use `inductor-A100-perf` as the data source for the dashboard, unless you are thinking to also upload data for some other purpose.

**Review by desertfire — CHANGES_REQUESTED** [review]

File: .github/workflows/upload-torch-dynamo-perf-stats.yml (hunk 1 above)
Comment (huydhn) [reply .github/workflows/upload-torch-dynamo-perf-stats.yml:5]: Noted.  I will remove the other twos.  There is no need to store the data if there is no immediate need to use them.  Adding them back is easy enough

Comment (huydhn) [reply .github/workflows/upload-torch-dynamo-perf-stats.yml]: In fact, this helps address my concern about the number of records to store. `inductor-A100-perf` runs very infrequently, so there will only be few hundred new records per day written to Rockset.

**Comment by weiwangmeta** [comment]
How do you plan to test this? I suggest reading from an already finished job's artifact, rather than running the entire workflow. 

On the same line:  do you need to wait for it to finish? https://github.com/pytorch/pytorch/actions/runs/4318195480

**Comment by huydhn** [comment]
> How do you plan to test this? I suggest reading from an already finished job's artifact, rather than running the entire workflow.

So far, I have been able to test this locally, i.e. `python -m tools.stats.upload_dynamo_perf_stats --workflow-run-id 4272892998 --workflow-run-attempt 1 --repo pytorch/pytorch`

This is hard to test this on the PR though because the workflow doesn't exist in trunk yet.  So I decide to just get it in first, and iterate on it if need.  I hope that the logic is simple enough that there is little chance of thing goes wrong.  And it's ok too as it doesn't break anything (yet)

> On the same line: do you need to wait for it to finish? https://github.com/pytorch/pytorch/actions/runs/4318195480

Yes, the workflow needs to run after perf workflow finishes, as the test report artifacts will only be available then on S3.

**Review by weiwangmeta — APPROVED** [review]
Looks awesome!

**Pull request closed without merging**